systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaTopicScalabilityUtils.java

```diff
@@ -0,0 +1,127 @@
+        KafkaTopicScalabilityUtils.checkTopicsState(topicPrefix, numberOfTopics, CustomResourceStatus.NotReady);
+    }
+
+    public static void checkTopicsReady(String topicPrefix, int numberOfTopics) {
+        LOGGER.info("Verifying that topics are in Ready state");
+        KafkaTopicScalabilityUtils.checkTopicsState(topicPrefix, numberOfTopics, CustomResourceStatus.Ready);
+    }
+
+    public static void checkTopicsContainConfig(String topicPrefix, int numberOfTopics, Map<String, Object> config) {
+        LOGGER.info("Verifying that topics contain right config");
+        List<CompletableFuture<?>> topics = new ArrayList<>();
+
+        for (int i = 0; i < numberOfTopics; i++) {
+            String currentTopic = topicPrefix + i;
+            topics.add(CompletableFuture.runAsync(() -> {
+                try {
+                    String json = cmdKubeClient(kubeClient().getNamespace()).exec("get", "kafkatopic", currentTopic, "-o", "jsonpath='{.spec.config}'").out();
+                    HashMap<?, ?> mapping = new ObjectMapper().readValue(json.replaceAll("'", ""), HashMap.class);
+                    assertEquals(mapping, config);
+                } catch (JsonProcessingException e) {
+                    fail(e.getMessage());
+                }
+            }));
+        }
+
+        CompletableFuture<Void> allTopics = CompletableFuture.allOf(topics.toArray(new CompletableFuture[0]))
+                .thenRun(() -> LOGGER.info("All topics contain right config"));
+
+        allTopics.join();
+    }
+
+    public static void modifyBigAmountOfTopics(String topicPrefix, int numberOfTopics, int numberOfPartitions, Map<String, Object> config) {
+        LOGGER.info("Modify topics via Kubernetes");
+        List<CompletableFuture<?>> topics = new ArrayList<>();
+
+        for (int i = 0; i < numberOfTopics; i++) {
+            String currentTopicName = topicPrefix + i;
+            LOGGER.info("Modify topic {}", currentTopicName);
+
+            topics.add(CompletableFuture.runAsync(() -> {
+                if (numberOfPartitions != 0) {
+                    KafkaTopicResource.replaceTopicResource(currentTopicName, kafkaTopic -> kafkaTopic.getSpec().setPartitions(numberOfPartitions));
+                }
+                if (!config.isEmpty()) {
+                    KafkaTopicResource.replaceTopicResource(currentTopicName, kafkaTopic -> kafkaTopic.getSpec().setConfig(config));
+                }
+            }));
+        }
+    }
+
+    public static void modifyBigAmountOfTopics(String topicPrefix, int numberOfTopics, int numberOfPartitions) {
+        KafkaTopicScalabilityUtils.modifyBigAmountOfTopics(topicPrefix, numberOfTopics, numberOfPartitions, new HashMap<>());
+    }
+
+    public static void modifyBigAmountOfTopics(String topicPrefix, int numberOfTopics, Map<String, Object> config) {
+        KafkaTopicScalabilityUtils.modifyBigAmountOfTopics(topicPrefix, numberOfTopics, 0, config);
+    }
+
```

Comment:
I guess you can pass `KafkaTopicSpec` here instead of doing some extra encapsulation. You will remove the `if` statements too.